src/Uno.UWP/System/Launcher.cs

```diff
@@ -27,18 +27,21 @@ public static IAsyncOperation<bool> LaunchUriAsync(Uri uri)
 			}
 
 #if !__WASM__
-			if (!CoreDispatcher.Main.HasThreadAccess)
+			if (CoreDispatcher.Main.HasThreadAccess)
 			{
-				if (typeof(Launcher).Log().IsEnabled(LogLevel.Error))
-				{
-					typeof(Launcher).Log().Error($"{nameof(LaunchUriAsync)} must be called on the UI thread");
-				}
-				// LaunchUriAsync throws the following exception if used on UI thread on UWP
```

Comment:
This comment looks very intentional. Have you verified the behavior on UWP and/or WinUI?

Reply:
Yes but kind of confusing:

>Unless you are calling this API from a Windows desktop application, this API must be called from within an ASTA thread (also known as the UI thread).

From the above, I understand that from UWP, when it's a desktop application, it can be called from other threads other than the UI Thread.

https://learn.microsoft.com/en-us/uwp/api/windows.system.launcher.launchuriasync?view=winrt-22621

Reply:
Let's make it work on non-UI thread for now. We may need to align it with the caller if possible in the future, particularly if there are multiple windows.